src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -431,6 +431,14 @@ public void filterForCourseDashboard(List<Participation> participations, String
         }
     }
 
+    public TutorParticipation getTutorParticipationForExercise(User tutor) {
+        Optional<TutorParticipation> tutorParticipation = this.tutorParticipations.stream()
```

Comment:
This sounds like an expensive operation. Remember, Hibernate would need to load all tutor participations and go through them. Could we add a database call in the corresponding repository instead?

Reply:
Good idea, since we have already the dedicated method